CMakeLists.txt

```diff
@@ -112,6 +112,14 @@ ecbuild_add_option( FEATURE GEOGRAPHY
     DESCRIPTION "Support for Geoiterator and nearest neighbour"
     DEFAULT ON )
 
+ecbuild_add_option( FEATURE ECKIT_GEO
+    DESCRIPTION "Support for Geoiterator and nearest neighbour (additional backend)"
+    CONDITION ENABLE_GEOGRAPHY
+    DEFAULT OFF )
+if( ECKIT_GEO AND NOT TARGET eckit_geo )
```

Comment:
This test did not work for me, so I pushed some changes, which now makes the detection of eckit and test of options work for me. Have a look and see what you think!

Reply:
I've changed the detection a little to use HAVE_ instead of ENABLE_ for handling ecbuild features, and directly testing the intended presence of the target (eckit_geo) as a required condition

Reply:
I do find it a bit confusing that there is mixed use of ecbuild_find_package and find_package, but that's I suppose something different to discuss